cmd/reviewdog/main.go

```diff
@@ -156,6 +158,7 @@ func init() {
 	flag.StringVar(&opt.level, "level", "error", levelDoc)
 	flag.BoolVar(&opt.guessPullRequest, "guess", false, guessPullRequestDoc)
 	flag.BoolVar(&opt.tee, "tee", false, teeDoc)
+	flag.BoolVar(&opt.filterByAddedLines, "filter-by-added-lines", false, filterByAddedLinesDoc)
```

Comment:
Can you create a string flag like enum instead of bool flag for extensibility?

e.g. `-filter-mode=[added or context or file]` where `added` is current behavior, `context` is filter results by diff "context" which can include unmodified lines as well (this p-r implementation).
`context` is ambiguous, so `diff`, `diff_context`, or something like that could be better. Do you have any suggestions?

`file` will be new implementation (probably I'll create it with another p-r) which reports results in changed files.
It won't work with github-pr-review reporter due to API restriction, but it should work for other reporters.

Reply:
@haya14busa sounds good. I like `diff_context` more because of explicitness.